### PR TITLE
The genesis block initialisation logic

### PR DESCRIPTION
**Pull request by gavofyork**
TODO:
- [x] Digest population for block authoring.

### CODE REVIEW

**Comment by arkpar** [comment]
Lots of `println!` in native code. Why not use logging macros instead?

File: wasm-runtime/polkadot/src/lib.rs [hunk 1]
```diff
@@ -48,9 +49,27 @@ pub fn execute_block(input: &[u8]) -> Vec<u8> {
...
-	let utx = UncheckedTransaction::from_slice(input).unwrap();
-	runtime::system::internal::execute_transaction(&utx);
+	let number = BlockNumber::from_slice(&input[0..8]).unwrap();
+	let utx = UncheckedTransaction::from_slice(&input[8..]).unwrap();
+	runtime::system::internal::execute_transaction(&utx, Header::from_block_number(number));
 	Vec::new()
 }
 
-impl_stubs!(execute_block, execute_transaction);
+/// Execute a given, serialised, transaction. Returns the empty vector.
```
Comment (arkpar) [inline wasm-runtime/polkadot/src/lib.rs]: Returns the header it seems

File: wasm-runtime/polkadot/src/primitives/header.rs [hunk 2]
```diff
@@ -37,6 +37,18 @@ pub struct Header {
 	pub digest: Digest,
 }
 
+impl Header {
+	pub fn from_block_number(number: BlockNumber) -> Self {
```
Comment (arkpar) [inline wasm-runtime/polkadot/src/primitives/header.rs]: This is for testing only, right? A doc comment would be ince

Comment (gavofyork) [reply wasm-runtime/polkadot/src/primitives/header.rs]: not really - useful in general when testing transactions without a specific preconstructed block to execute them within. doc comment sure

**Pull request merged**